flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  description = "An efficient, portable erasure coding tool";
+
+  inputs = {
+    # Nix Inputs
+    nixpkgs.url = github:nixos/nixpkgs/?ref=nixos-22.11;
+    flake-utils.url = github:numtide/flake-utils;
+    hs-flake-utils.url = "git+https://whetstone.private.storage/jcalderone/hs-flake-utils.git?ref=main";
```

Comment:
Feels like I can't do a good review without reading this reference, but I'll skip it for now.